spec/controllers/event/lists_controller_spec.rb

```diff
@@ -10,14 +10,148 @@
 
 describe Event::ListsController do
 
-  before { sign_in(people(:top_leader)) }
+  before { sign_in(user) }
   render_views
 
   let(:bsv) { CSV.parse(response.body, headers: true) }
   let(:kind) { event_kinds(:flock) }
+  let(:user) { people(:top_leader) }
+  let(:now) { Time.zone.now }
+  subject { assigns(:camps).values.flatten }
 
   it 'GET#courses with kind csv' do
     get :bsv_export, params: { filter: { kinds: kind.id.to_s, bsv_since: '1.1.2012', bsv_until: '31.12.2012', states: ['closed'] } }
     expect(bsv).to have(0).item
   end
+
+  describe 'GET all_camps' do
+    let!(:current) do
+      current = Fabricate(:camp)
+      current.dates = [Fabricate(:event_date, start_at: now - 15.days, finish_at: now - 10.days),
+                       Fabricate(:event_date, start_at: now - 5.days, finish_at: now + 5.days)]
+      current
+    end
+
+    let!(:upcoming_in_range) do
+      upcoming_in_range = Fabricate(:camp)
+      upcoming_in_range.dates = [Fabricate(:event_date, start_at: now + 10.days, finish_at: nil),
+                                 Fabricate(:event_date, start_at: now + 12.days, finish_at: nil)]
+      upcoming_in_range
+    end
+
+    let!(:upcoming_outside_range) do
+      upcoming_outside_range = Fabricate(:camp)
+      upcoming_outside_range.dates = [Fabricate(:event_date, start_at: now + 50.days, finish_at: now + 55.days)]
+      upcoming_outside_range
```

Comment:
Funktioniert vielleicht das hier?

```suggestion
      Fabricate(:camp, dates: [
        Fabricate(:event_date, start_at: now + 50.days, finish_at: now + 55.days)
      ])
```

Oder nofalls sowas:

```suggestion
      Fabricate(:camp).tap do |camp|
        camp.dates = [Fabricate(:event_date, start_at: now + 50.days, finish_at: now + 55.days)]
      end
```

Wenn nicht, auch okay. Es scheint nur so unnötig, das für weniger als ein Blinzeln in einer Variable zu speichern.